llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -6544,6 +6544,35 @@ bool InstCombinerImpl::replacedSelectWithOperand(SelectInst *SI,
   return false;
 }
 
+static std::optional<bool> compareKnownBits(ICmpInst::Predicate Pred,
```

Comment:
We already implemented similar logic in `CombinerHelper::matchICmpToTrueFalseKnownBits`. Do we need a common helper?

Reply:
Opened https://github.com/llvm/llvm-project/pull/115878 to add the common helper.

Reply:
Now using the common helper.